File: 0733-flood-fill/0733-flood-fill.cpp
class Solution {
public:

void colour(vector<vector<int>>& image, int r, int c, int color,int elem){
   if(r<0 || c<0 || r== image.size() || c== image[0].size() || image[r][c] != elem){
     return;
   }
   if(image[r][c] == elem){
    image[r][c] = color;
   }

 
  colour(image,r+1,c,color,elem);
  colour(image,r-1,c,color,elem);
  colour(image,r,c+1,color,elem);
  colour(image,r,c-1,color,elem);
}
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        int elem = image[sr][sc];
        if(elem == color){
            return image;
        }
        colour(image,sr,sc,color,elem);
        return image;
    }
};

File: 0733-flood-fill/0733-flood-fill.java
class Solution {
    void DFS(int i , int j , int[][]image,int color , int src){
        image[i][j] = color;
        int[] rows = {0,0,-1,+1};
        int[] cols = {-1,+1,0,0};

        for(int idx=0;idx<=3;idx++){
          int r = i + rows[idx];
          int c = j + cols[idx];

          if(r >=0 && r <image.length && c >=0 && c < image[0].length && image[r][c] == src){
            
            DFS(r,c,image,color,src);
          } 
        }
    }
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if(sr<0 || sr >= image.length || sc < 0 || sc>=image[0].length) return image;
        int src = image[sr][sc];
        if(src == color) return image;
        DFS(sr,sc,image,color,src);

        return image;
    }
}